Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -138,5 +146,142 @@ lemma map [TopologicalSpace 𝓨] [MeasurableSpace 𝓨] [OpensMeasurableSpace 
+    ∃ (k : ℕ), ∀ μ ∈ S, 1 - ε < μ (⋃ i ≤ k, U i) := by
+  have εfin : ε ≠ ∞ := ne_top_of_le_ne_top (by simp) hεbound
+  lift ε to ℝ≥0 using εfin
+  obtain ⟨ε, hε'⟩ := ε
+  simp only [ENNReal.coe_pos, ← NNReal.coe_lt_coe, NNReal.coe_zero, coe_mk, coe_le_one_iff,
+      ← NNReal.coe_le_coe, NNReal.coe_one] at hε hεbound
+  by_contra! nh
+  choose μ hμInS hcontradiction using nh
+  obtain ⟨μlim, _, sub, hsubmono, hμconverges⟩ :=
+      hcomp.isSeqCompact (fun n ↦ subset_closure <| hμInS n)
+  have Measurebound n : (μlim (⋃ (i ≤ n), U i) : ℝ) ≤ 1 - ε := calc
+    (μlim (⋃ (i ≤ n), U i) : ℝ)
+    _ ≤ liminf (fun k ↦ (μ (sub k) (⋃ (i ≤ n), U i) : ℝ)) atTop := by
+      have hopen : IsOpen (⋃ i ≤ n, U i) := isOpen_biUnion fun i a ↦ O i
+      have := ProbabilityMeasure.le_liminf_measure_open_of_tendsto hμconverges hopen
+      simp_rw [Function.comp_apply, ← ProbabilityMeasure.ennreal_coeFn_eq_coeFn_toMeasure] at this
+      rw [← ofNNReal_liminf] at this
+      · exact mod_cast this
+      use 1
+      simpa [ge_iff_le, eventually_map, eventually_atTop, forall_exists_index] using fun _ x h ↦
+          (h x (by simp)).trans <| ProbabilityMeasure.apply_le_one (μ (sub x)) (⋃ i ≤ n, U i)
+    _ ≤ liminf (fun k ↦ (μ (sub k) (⋃ (i ≤ sub k), U i) : ℝ)) atTop := by
+      apply Filter.liminf_le_liminf
+      · simp only [NNReal.coe_le_coe, eventually_atTop, ge_iff_le]
+        use n + 1
+        intro b hypo
+        refine (μ (sub b)).apply_mono
+            <| Set.biUnion_mono (fun i (hi : i ≤ n) ↦ hi.trans ?_) fun _ _ ↦ le_rfl
+        exact le_trans (Nat.le_add_right n 1) (le_trans hypo (StrictMono.le_apply hsubmono))
+      · use 0; simp
+      · use 1
+        simpa [ge_iff_le, eventually_map, eventually_atTop, ge_iff_le, forall_exists_index] using
+            fun _ d hyp ↦ (hyp d (by simp)).trans (by simp)
+    _ ≤ 1 - ε := by
+      apply Filter.liminf_le_of_le
+      · use 0; simp
+      simp only [eventually_atTop, ge_iff_le, forall_exists_index]
+      intro b c h
+      apply le_trans (h c le_rfl)
+      refine (ofReal_le_ofReal_iff (by rw [sub_nonneg]; exact hεbound)).mp ?_
+      rw [ofReal_coe_nnreal]
+      apply le_trans (hcontradiction (sub c))
+      norm_cast
+  have accumulation : Tendsto (fun n ↦ μlim (⋃ i ≤ n, U i)) atTop (𝓝 (μlim (⋃ i, U i))) := by
+    simp_rw [← Set.accumulate_def, ProbabilityMeasure.tendsto_measure_iUnion_accumulate]
+  rw [Cov, coeFn_univ, ← NNReal.tendsto_coe] at accumulation
+  have exceeds_bound : ∀ᶠ n in atTop, (1 - ε / 2 : ℝ) ≤ μlim (⋃ i ≤ n, U i) :=
+      Tendsto.eventually_const_le (v := 1) (by simp only [sub_lt_self_iff, Nat.ofNat_pos,
+                    div_pos_iff_of_pos_right]; positivity) accumulation
+  suffices ∀ᶠ n : ℕ in atTop, False from this.exists.choose_spec
+  filter_upwards [exceeds_bound] with n hn
+  linarith [hn.trans <| Measurebound n]
+
+variable [CompleteSpace 𝓧]
+
+/-In a second countable complete metric space, a precompact set of probability measures is tight.-/
```

Comment:
That's a regular comment /- -/, not a docstring /-- -/
If you don't mind, I'd also like to replace precompact, which is not a word used in the code (and is sometimes used for another meaning), by the more explicit "with compact closure".

```suggestion
/-- In a second countable complete metric space, a set of probability measures with compact closure
is tight. -/
```